web-client/extras/lib/SerialBuffer.ts

```diff
@@ -0,0 +1,205 @@
+        } else if (value === 0xFE) {
+            return this.readUint32();
+        } /*if (value === 0xFF)*/ else {
+            return this.readUint64();
+        }
+    }
+
+    writeVarUint(value: number): void {
+        if (!NumberUtils.isUint64(value)) throw new Error('Malformed value');
+        if (value < 0xFD) {
+            this.writeUint8(value);
+        } else if (value <= 0xFFFF) {
+            this.writeUint8(0xFD);
+            this.writeUint16(value);
+        } else if (value <= 0xFFFFFFFF) {
+            this.writeUint8(0xFE);
+            this.writeUint32(value);
+        } else {
+            this.writeUint8(0xFF);
+            this.writeUint64(value);
+        }
+    }
+
+    static varUintSize(value: number): number {
+        if (!NumberUtils.isUint64(value)) throw new Error('Malformed value');
+        if (value < 0xFD) {
+            return 1;
+        } else if (value <= 0xFFFF) {
+            return 3;
+        } else if (value <= 0xFFFFFFFF) {
+            return 5;
+        } else {
+            return 9;
+        }
+    }
+
+    readFloat64(): number {
+        const value = this._view.getFloat64(this._readPos);
+        this._readPos += 8;
+        return value;
+    }
+
+    writeFloat64(value: number): void {
+        this._view.setFloat64(this._writePos, value);
+        this._writePos += 8;
+    }
+
+    readString(length: number): string {
+        const bytes = this.read(length);
+        return BufferUtils.toAscii(bytes);
+    }
+
+    writeString(value: string, length: number): void {
+        if (StringUtils.isMultibyte(value) || value.length !== length) throw new Error('Malformed value/length');
+        const bytes = BufferUtils.fromAscii(value);
+        this.write(bytes);
+    }
+
+    readPaddedString(length: number): string {
+        const bytes = this.read(length);
+        let i = 0;
+        while (i < length && bytes[i] !== 0x0) i++;
+        const view = new Uint8Array(bytes.buffer, bytes.byteOffset, i);
+        return BufferUtils.toAscii(view);
+    }
+
+    writePaddedString(value: string, length: number): void {
+        if (StringUtils.isMultibyte(value) || value.length > length) throw new Error('Malformed value/length');
+        const bytes = BufferUtils.fromAscii(value);
+        this.write(bytes);
+        const padding = length - bytes.byteLength;
+        this.write(new Uint8Array(padding));
+    }
+
+    readVarLengthString(): string {
+        const length = this.readUint8();
+        if (this._readPos + length > this.length) throw new Error('Malformed length');
+        const bytes = this.read(length);
+        return BufferUtils.toAscii(bytes);
+    }
+
+    writeVarLengthString(value: string): void {
+        if (StringUtils.isMultibyte(value) || !NumberUtils.isUint8(value.length)) throw new Error('Malformed value');
+        const bytes = BufferUtils.fromAscii(value);
+        this.writeUint8(bytes.byteLength);
+        this.write(bytes);
+    }
```

Comment:
That's not how we do variable-length strings in the codebase.

Reply:
How do we do it then? With a varint length?

Reply:
Yes, with varint length: https://postcard.jamesmunns.com/wire-format#15---string.

Reply:
Done in https://github.com/nimiq/core-rs-albatross/pull/2715/commits/3103f08eaf93752c0376f93e25e2b079f81aaae1